client/cypress/integration/visualizations/chart_spec.js

```diff
@@ -0,0 +1,204 @@
+/* global cy */
+
+import { getWidgetTestId } from "../../support/dashboard";
+
+const SQL = `
+  SELECT 'a' AS stage, 11 AS value1, 22 AS value2 UNION ALL
+  SELECT 'a' AS stage, 12 AS value1, 41 AS value2 UNION ALL
+  SELECT 'a' AS stage, 45 AS value1, 93 AS value2 UNION ALL
+  SELECT 'a' AS stage, 54 AS value1, 79 AS value2 UNION ALL
+  SELECT 'b' AS stage, 33 AS value1, 65 AS value2 UNION ALL
+  SELECT 'b' AS stage, 73 AS value1, 50 AS value2 UNION ALL
+  SELECT 'b' AS stage, 90 AS value1, 40 AS value2 UNION ALL
+  SELECT 'c' AS stage, 19 AS value1, 33 AS value2 UNION ALL
+  SELECT 'c' AS stage, 92 AS value1, 14 AS value2 UNION ALL
+  SELECT 'c' AS stage, 63 AS value1, 65 AS value2 UNION ALL
+  SELECT 'c' AS stage, 44 AS value1, 27 AS value2\
+`;
+
+/**
+ * Asserts the preview canvas exists, then captures the g.points element, which should be generated by plotly and asserts whether it exists
+ * @param should Passed to should expression after plot points are captured
+ */
+function assertPlotPreview(should = "exist") {
```

Comment:
One quick note: those helper functions should be in a support file, similar to `client/cypress/support/visualizations/table.js`. This way we will have it a bit simplified in here, with code more specific to the tests we are performing.